augur/tasks/util/collection_util.py

```diff
@@ -425,3 +433,155 @@ def start_block_of_repos(logger,session,condition,limit,phases,repos_type,hook="
+
+    #getattr(CollectionStatus,f"{hook}_status" ) represents the status of the given hook
+    #Get the count of repos that are currently running this collection hook
+    desired_status = f"{hook}_status"
+    active_repo_count = len(session.query(CollectionStatus).filter(getattr(CollectionStatus,desired_status ) == CollectionState.COLLECTING.value).all())
+
+    #Will always disallow errored repos and repos that are already collecting
+    not_erroed = getattr(CollectionStatus,desired_status ) != str(CollectionState.ERROR.value)
+    not_collecting = getattr(CollectionStatus,desired_status ) != str(CollectionState.COLLECTING.value)
+
+    #The maximum amount of repos to schedule is affected by the existing repos running tasks
+    limit = max_repo-active_repo_count
+
+    #Split all users that have new repos into four lists and randomize order
+    query = s.sql.text(f"""
+        SELECT  
+        user_id
+        FROM augur_operations.user_groups 
+        JOIN augur_operations.user_repos ON augur_operations.user_groups.group_id = augur_operations.user_repos.group_id
+        JOIN augur_data.repo ON augur_operations.user_repos.repo_id = augur_data.repo.repo_id
+        JOIN augur_operations.collection_status ON augur_operations.user_repos.repo_id = augur_operations.collection_status.repo_id
+        WHERE {desired_status}='{str(new_status)}'
+        GROUP BY user_id
+    """)
+
+    user_list = session.execute_sql(query).fetchall()
+    random.shuffle(user_list)
+
+    #Extract the user id from the randomized list and split into four chunks
+    split_user_list = split_list_into_chunks([row[0] for row in user_list], 4)
+
+    session.logger.info(f"User_list: {split_user_list}")
+
+    #Iterate through each fourth of the users fetched
+    for quarter_list in split_user_list:
+        if limit <= 0:
+            return
+
+        #Query a set of valid repositories sorted by weight, also making sure that the repos are new
+        repo_query = s.sql.text(f"""
+            SELECT
+	        repo.repo_id
+            FROM augur_operations.user_groups 
+            JOIN augur_operations.user_repos ON augur_operations.user_groups.group_id = augur_operations.user_repos.group_id
+            JOIN augur_data.repo ON augur_operations.user_repos.repo_id = augur_data.repo.repo_id
+            JOIN augur_operations.collection_status ON augur_operations.user_repos.repo_id = augur_operations.collection_status.repo_id
+            WHERE user_id IN :list_of_user_ids AND {desired_status}='{str(new_status)}'
+            ORDER BY {hook}_weight
+            LIMIT :limit_num
+        """).bindparams(list_of_user_ids=tuple(quarter_list),limit_num=(limit*2))
+
+        #Get a list of valid repo ids, limit set to 2 times the usual
+        valid_repos = session.execute_sql(repo_query).fetchall()
+        valid_repo_git_ids = [repo[0] for repo in valid_repos]
+
```

Comment:
I think `valid_repo_ids` would be a more descriptive, `valid_repo_git_ids` isn't super clear to me, because the ids are of the repo not the `repo_git`